### ISSUE TITLE
Code component with taxindex=0 should have role and accessible name

### ISSUE THREAD

**Issue by straker** [dequelabs/cauldron#680]
The `Code` component allows adding `tabindex=0` to support keyboard only scrolling of the code. When tabindex is applied, it should also [have a role and accessible name](https://adrianroselli.com/2022/06/keyboard-only-scrolling-areas.html) so screen readers know what the element in the tab order is doing. `role=region` is the typical role to add, but doing so also requires that the element [have an accessible name](https://www.w3.org/TR/wai-aria-1.2/#region). 

When `tabindex` is used on the `Code` component, it should add `role=region` to the element and require that the element have an accessible name (either `aria-label` or `aria-labelledby`) as `region` role is not named from contents.
